scripts/check_format.sh

```diff
@@ -1,6 +1,18 @@
 #!/bin/bash
 
-set -eo pipefail
+set -e # Stop only for critical failures
+
+echo "Setting up format check..."
+
+# Detect OS and run the appropriate setup script
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  bash scripts/ci/ubuntu/0-setup.sh
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  bash scripts/ci/macos/0-setup.sh
+else
+  echo "⚠️ Unsupported OS: $OSTYPE"
+  exit 1
+fi
```

Comment:
We don't like branches when we can avoid one. We can expect sequencing at the call-site (say your MacBook, or another developer's Linux machine) to do either one of:

(CI Ubuntu) 

```
bash scripts/ci/ubuntu/0-setup.sh
bash scripts/format-check.sh
```

Your mac:
```
bash scripts/ci/macos/0-setup.sh
bash scripts/format-check.sh
```